### PR TITLE
[SPARK-33296][SQL] Format exception of unsupported nested grouping analytics

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Nested grouping. analytics can't run but throw  ambiguous exception, This pr make these case throw exceptions that are easy for users to understand
```
spark-sql>  select a, b, c from x group by cube(a, b) with cube;;
20/10/30 12:20:20 ERROR SparkSQLDriver: Failed in [ select a, b, c from x group by cube(a, b) with cube]
java.lang.UnsupportedOperationException
	at org.apache.spark.sql.catalyst.expressions.GroupingSet.dataType(grouping.scala:36)
	at org.apache.spark.sql.catalyst.expressions.GroupingSet.dataType$(grouping.scala:36)
	at org.apache.spark.sql.catalyst.expressions.Cube.dataType(grouping.scala:61)


spark-sql> select a, b, c from x group by cube(a, b, c) grouping sets(a, b, (a, b));
Error in query: unresolved operator 'GroupingSets [ArrayBuffer(a#0), ArrayBuffer(b#1), ArrayBuffer(a#0, b#1)], [cube(a#0, b#1, c#2)], [a#0, b#1, c#2];;
'GroupingSets [ArrayBuffer(a#0), ArrayBuffer(b#1), ArrayBuffer(a#0, b#1)], [cube(a#0, b#1, c#2)], [a#0, b#1, c#2]
+- SubqueryAlias spark_catalog.default.x
   +- HiveTableRelation [`default`.`x`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#0, b#1, c#2], Partition Cols: []]


spark-sql>  select a, b, c from x group by a, b,  c grouping sets(a, b, cube(a, b));
Error in query: unresolved operator 'GroupingSets [ArrayBuffer(a#3), ArrayBuffer(b#4), ArrayBuffer(cube(a#3, b#4))], [a#3, b#4, c#5], [a#3, b#4, c#5];;
'GroupingSets [ArrayBuffer(a#3), ArrayBuffer(b#4), ArrayBuffer(cube(a#3, b#4))], [a#3, b#4, c#5], [a#3, b#4, c#5]
+- SubqueryAlias spark_catalog.default.x
   +- HiveTableRelation [`default`.`x`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#3, b#4, c#5], Partition Cols: []]

```

### Why are the changes needed?
Throw an exception that is easy for the user to understand

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Add UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
FYI @maropu

**Comment by AngersZhuuuu** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala [hunk 1]
```diff
@@ -3691,6 +3691,49 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
...
+      }
+
+      sql("CREATE TABLE t(a int, b int, c int) USING PARQUET")
+      checkAnalysisException(
+        "SELECT a, b, count(c) FROM t GROUP BY cube(a, b) WITH CUBE",
+        "Use grouping analytics function like `cube|rollup` with " +
+          "`WITH CUBE` and `WITH ROLLUP` is not supported."
+      )
+      checkAnalysisException(
+        "SELECT a, b, count(c) FROM t GROUP BY rollup(a, b) WITH CUBE",
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala:3711]: If you refactor the parsing logic for `GROUP BY`, I think the parser cannot accept this case. Right? If so, I think we don't need to update the Analyzer code.

Comment (AngersZhuuuu) [reply sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala:3711]: > If you refactor the parsing logic for `GROUP BY`, I think the parser cannot accept this case. Right? If so, I think we don't need to update the Analyzer code.

Ideally, we can throw such an exception at the Parser layer.

**Pull request closed without merging**